File: src/types/domain.ts
import { z } from "zod";

const nullableText = z.string().nullable();

export const writingSystemSchema = z.object({
  id: z.string(),
  name: z.string(),
  type: z.enum([
    "orthography",
    "romanization",
    "transliteration",
    "phonemic",
    "phonetic",
    "other",
  ]),
  scriptCode: nullableText,
  languageTag: nullableText,
  displayRole: z.enum(["primary", "secondary"]).nullable(),
  sortOrder: z.number(),
  fontFamily: nullableText,
  notes: nullableText,
});

export const entryFormSchema = z.object({
  id: z.string(),
  writingSystemId: z.string(),
  text: z.string(),
  variantLabel: nullableText,
  dialect: nullableText,
  status: nullableText,
  notes: nullableText,
  sortOrder: z.number(),
});

export const exampleFormSchema = z.object({
  id: z.string(),
  writingSystemId: z.string(),
  text: z.string(),
  sortOrder: z.number(),
});

export const exampleSchema = z.object({
  id: z.string(),
  translation: nullableText,
  notes: nullableText,
  sortOrder: z.number(),
  forms: z.array(exampleFormSchema),
});

export const senseSchema = z.object({
  id: z.string(),
  gloss: nullableText,
  definition: nullableText,
  partOfSpeech: nullableText,
  semanticDomain: nullableText,
  sortOrder: z.number(),
  examples: z.array(exampleSchema),
});

export const senseImageSchema = z.object({
  id: z.string(),
  senseId: z.string(),
  originalFilename: z.string(),
  width: z.number().int().positive(),
  height: z.number().int().positive(),
  byteSize: z.number().int().positive(),
  sortOrder: z.number().int(),
  createdAt: z.string(),
});

export const relationSchema = z.object({
  id: z.string(),
  targetEntryId: nullableText,
  relationType: z.enum(["root", "base"]),
  fallbackText: nullableText,
  notes: nullableText,
  sortOrder: z.number(),
});

export const lexicalEntrySchema = z.object({
  id: z.string(),
  notes: nullableText,
  sectionOverride: nullableText,
  revision: z.number(),
  createdAt: z.string(),
  updatedAt: z.string(),
  forms: z.array(entryFormSchema),
  senses: z.array(senseSchema),
  relations: z.array(relationSchema),
});

export const entrySummarySchema = z.object({
  id: z.string(),
  primaryForm: z.string(),
  secondaryForm: nullableText,
  pronunciationForm: nullableText,
  pronunciationWritingSystemId: nullableText,
  senses: z.array(z.object({
    partOfSpeech: nullableText,
    gloss: nullableText,
  })),
  revision: z.number(),
  sectionLabel: nullableText,
  manualOrderPending: z.boolean(),
});

export const entrySortSettingsSchema = z.object({
  version: z.literal(2),
  mode: z.enum(["auto", "manual"]),
  source: z.enum(["writingSystem", "semanticDomain"]),
  writingSystemId: z.string(),
  alphabet: z.array(z.string()),
});

export const manualSortItemSchema = z.discriminatedUnion("kind", [
  z.object({ kind: z.literal("heading"), id: z.string(), label: z.string() }),
  z.object({ kind: z.literal("entry"), entryId: z.string() }),
]);

export const manualSortLayoutSchema = z.object({ version: z.literal(1), items: z.array(manualSortItemSchema) });

export const corpusPartOfSpeechSchema = z.enum([
  "noun", "verb", "adjective", "adverb", "pronoun", "particle", "other",
]);

export const fontPresetSchema = z.enum([
  "auto", "charisSil", "notoSerif", "notoSerifCjkTc", "chironSungHk", "chironHeiHk",
]);

export const exportSettingsSchema = z.object({
  version: z.literal(1),
  corpus: z.object({
    partOfSpeechMappings: z.record(z.string(), corpusPartOfSpeechSchema),
  }),
  latex: z.object({
    title: z.string(),
    author: z.string(),
    headwordWritingSystemId: z.string(),
    pronunciationWritingSystemId: nullableText,
    exampleWritingSystemId: z.string(),
    collationLanguageTag: nullableText,
    sectionMode: z.enum(["auto", "firstGrapheme", "none"]),
    reverseIndex: z.enum(["gloss", "none"]),
    relatedEntries: z.enum(["none", "root", "base", "both"]),
    includeSenseImages: z.boolean(),
    includeSemanticDomains: z.boolean().default(true),
    fontPresets: z.record(z.string(), fontPresetSchema),
  }),
});

export const exportKindSchema = z.enum(["corpusCsv", "latex", "pdf"]);
export const exportIssueSchema = z.object({
  severity: z.enum(["error", "warning"]),
  code: z.string(),
  entryId: nullableText,
  senseId: nullableText,
  field: nullableText,
  details: nullableText,
});
export const fontPackStatusSchema = z.object({
  id: z.string(),
  version: z.string(),
  state: z.enum(["missing", "installed", "invalid"]),
  mandatory: z.boolean(),
  installedBytes: z.number(),
});
export const fontInstallProgressSchema = z.object({
  packId: z.string(),
  phase: z.enum(["downloading", "verifying", "installed", "failed"]),
  packIndex: z.number().int().nonnegative(),
  packCount: z.number().int().nonnegative(),
  downloadedBytes: z.number().nonnegative(),
  totalBytes: z.number().nonnegative().nullable(),
});
export const csvDelimiterSchema = z.enum(["comma", "tab", "semicolon"]);
export const csvColumnSchema = z.object({ index: z.number().int().nonnegative(), name: z.string(), samples: z.array(z.string()) });
export const csvInspectionSchema = z.object({
  sourcePath: z.string(), fileName: z.string(), sha256: z.string(), delimiter: csvDelimiterSchema,
  rowCount: z.number().int().nonnegative(), columns: z.array(csvColumnSchema), profile: nullableText,
});
export const csvMappingTargetSchema = z.discriminatedUnion("kind", [
  z.object({ kind: z.literal("ignore") }),
  z.object({ kind: z.literal("entryForm"), writingSystemId: z.string() }),
  z.object({ kind: z.literal("entryNotes") }),
  z.object({ kind: z.literal("senseGloss") }),
  z.object({ kind: z.literal("senseDefinition") }),
  z.object({ kind: z.literal("partOfSpeech") }),
  z.object({ kind: z.literal("semanticDomain") }),
  z.object({ kind: z.literal("exampleForm"), writingSystemId: z.string() }),
  z.object({ kind: z.literal("exampleTranslation") }),
  z.object({ kind: z.literal("exampleNotes") }),
  z.object({ kind: z.literal("rootFallback") }),
]);
export const csvColumnMappingSchema = z.object({ columnIndex: z.number().int().nonnegative(), target: csvMappingTargetSchema });
export const csvImportGroupSchema = z.object({ rowIndices: z.array(z.number().int().nonnegative()) });
export const csvProjectSpecSchema = z.object({
  parentDir: z.string(), name: z.string(), languageName: nullableText, languageCode: nullableText,
  analysisLanguage: z.enum(["zh-TW", "en"]).nullable(), writingSystems: z.array(writingSystemSchema),
});
export const csvPreviewRequestSchema = z.object({
  sourcePath: z.string(), delimiter: csvDelimiterSchema, project: csvProjectSpecSchema,
  mappings: z.array(csvColumnMappingSchema), groups: z.array(csvImportGroupSchema),
  excludedRows: z.array(z.number().int().nonnegative()), rootDelimiter: z.string(),
});
export const csvPreviewIssueSchema = z.object({
  severity: z.enum(["error", "warning"]), code: z.string(),
  rowIndices: z.array(z.number().int().nonnegative()),
  columnIndices: z.array(z.number().int().nonnegative()).default([]), details: nullableText,
});
export const csvImportPreviewSchema = z.object({
  previewToken: z.string(), sourceRowCount: z.number().int().nonnegative(), importEntryCount: z.number().int().nonnegative(),
  importSenseCount: z.number().int().nonnegative(), skippedRowCount: z.number().int().nonnegative(),
  blockingErrorCount: z.number().int().nonnegative(), warningCount: z.number().int().nonnegative(),
  issues: z.array(csvPreviewIssueSchema), groups: z.array(z.object({ rowIndices: z.array(z.number().int().nonnegative()), primaryForm: z.string(), blocked: z.boolean() })),
});
export const exportPreviewSchema = z.object({
  snapshotToken: z.string(),
  rowCount: z.number(),
  issues: z.array(exportIssueSchema),
  omitted: z.object({
    examples: z.number(),
    exampleForms: z.number(),
    baseRelations: z.number(),
  }),
  requiredFontPacks: z.array(fontPackStatusSchema),
});
export const exportResultSchema = z.object({
  csvPath: nullableText,
  latexDirectory: nullableText,
  zipPath: nullableText,
  pdfPath: nullableText,
  pdfStatus: z.enum(["notRequested", "created", "xeLatexMissing", "failed"]),
  rowCount: z.number(),
  issues: z.array(exportIssueSchema),
  diagnosticPath: nullableText,
});
export const texEngineStatusSchema = z.object({
  available: z.boolean(),
  path: nullableText,
});

export const senseImageMutationSchema = z.object({
  entry: lexicalEntrySchema,
  image: senseImageSchema.nullable(),
});

export const senseImageContentSchema = z.object({
  mimeType: z.literal("image/png"),
  dataBase64: z.string(),
});

export const projectSchema = z.object({
  id: z.string(),
  name: z.string(),
  languageName: nullableText,
  languageCode: nullableText,
  analysisLanguage: z.enum(["zh-TW", "en"]).nullable(),
  description: nullableText,
  createdAt: z.string(),
  updatedAt: z.string(),
});

export const projectSnapshotSchema = z.object({
  rootPath: z.string(),
  project: projectSchema,
  writingSystems: z.array(writingSystemSchema),
  partOfSpeechOptions: z.array(z.string()),
  semanticDomainOptions: z.array(z.string()),
  exportSettings: exportSettingsSchema,
  entrySortSettings: entrySortSettingsSchema,
  manualSortLayout: manualSortLayoutSchema,
  entries: z.array(entrySummarySchema),
});
export const csvImportResultSchema = z.object({
  snapshot: projectSnapshotSchema, importedEntryCount: z.number().int().nonnegative(),
  importedSenseCount: z.number().int().nonnegative(), skippedRowCount: z.number().int().nonnegative(),
  warnings: z.array(csvPreviewIssueSchema),
});

export const deletedEntrySchema = z.object({
  id: z.string(),
  deletedAt: z.string(),
});

export type WritingSystem = z.infer<typeof writingSystemSchema>;
export type EntryForm = z.infer<typeof entryFormSchema>;
export type ExampleForm = z.infer<typeof exampleFormSchema>;
export type Example = z.infer<typeof exampleSchema>;
export type Sense = z.infer<typeof senseSchema>;
export type SenseImage = z.infer<typeof senseImageSchema>;
export type EntryRelation = z.infer<typeof relationSchema>;
export type LexicalEntry = z.infer<typeof lexicalEntrySchema>;
export type EntrySummary = z.infer<typeof entrySummarySchema>;
export type EntrySortSettings = z.infer<typeof entrySortSettingsSchema>;
export type ManualSortItem = z.infer<typeof manualSortItemSchema>;
export type ManualSortLayout = z.infer<typeof manualSortLayoutSchema>;
export type Project = z.infer<typeof projectSchema>;
export type ProjectSnapshot = z.infer<typeof projectSnapshotSchema>;
export type CorpusPartOfSpeech = z.infer<typeof corpusPartOfSpeechSchema>;
export type ExportSettings = z.infer<typeof exportSettingsSchema>;
export type ExportKind = z.infer<typeof exportKindSchema>;
export type ExportIssue = z.infer<typeof exportIssueSchema>;
export type ExportPreview = z.infer<typeof exportPreviewSchema>;
export type ExportResult = z.infer<typeof exportResultSchema>;
export type TexEngineStatus = z.infer<typeof texEngineStatusSchema>;
export type FontPackStatus = z.infer<typeof fontPackStatusSchema>;
export type FontInstallProgress = z.infer<typeof fontInstallProgressSchema>;
export type CsvDelimiter = z.infer<typeof csvDelimiterSchema>;
export type CsvInspection = z.infer<typeof csvInspectionSchema>;
export type CsvMappingTarget = z.infer<typeof csvMappingTargetSchema>;
export type CsvColumnMapping = z.infer<typeof csvColumnMappingSchema>;
export type CsvImportGroup = z.infer<typeof csvImportGroupSchema>;
export type CsvProjectSpec = z.infer<typeof csvProjectSpecSchema>;
export type CsvPreviewRequest = z.infer<typeof csvPreviewRequestSchema>;
export type CsvPreviewIssue = z.infer<typeof csvPreviewIssueSchema>;
export type CsvImportPreview = z.infer<typeof csvImportPreviewSchema>;
export type CsvImportResult = z.infer<typeof csvImportResultSchema>;

export function createId(): string {
  return crypto.randomUUID();
}

File: src-tauri/src/domain.rs
use std::collections::BTreeMap;

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub name: String,
    pub language_name: Option<String>,
    pub language_code: Option<String>,
    pub analysis_language: Option<String>,
    pub description: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum CorpusPartOfSpeech {
    Noun,
    Verb,
    Adjective,
    Adverb,
    Pronoun,
    Particle,
    Other,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct CorpusExportSettings {
    pub part_of_speech_mappings: BTreeMap<String, CorpusPartOfSpeech>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum SectionMode {
    Auto,
    FirstGrapheme,
    None,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum ReverseIndexMode {
    Gloss,
    None,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum FontPreset {
    Auto,
    CharisSil,
    NotoSerif,
    NotoSerifCjkTc,
}

#[derive(Debug, Clone, Copy, Default, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum RelatedEntriesMode {
    #[default]
    None,
    Root,
    Base,
    Both,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct LatexExportSettings {
    pub title: String,
    pub author: String,
    pub headword_writing_system_id: String,
    pub pronunciation_writing_system_id: Option<String>,
    pub example_writing_system_id: String,
    pub collation_language_tag: Option<String>,
    pub section_mode: SectionMode,
    pub reverse_index: ReverseIndexMode,
    #[serde(default)]
    pub related_entries: RelatedEntriesMode,
    pub font_presets: BTreeMap<String, FontPreset>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ExportSettingsV1 {
    pub version: u8,
    pub corpus: CorpusExportSettings,
    pub latex: LatexExportSettings,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum ExportKind {
    CorpusCsv,
    Latex,
    Pdf,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum ExportIssueSeverity {
    Error,
    Warning,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ExportIssue {
    pub severity: ExportIssueSeverity,
    pub code: String,
    pub entry_id: Option<String>,
    pub sense_id: Option<String>,
    pub field: Option<String>,
    pub details: Option<String>,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct OmittedExportData {
    pub examples: usize,
    pub example_forms: usize,
    pub base_relations: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ExportPreview {
    pub snapshot_token: String,
    pub row_count: usize,
    pub issues: Vec<ExportIssue>,
    pub omitted: OmittedExportData,
    pub required_font_packs: Vec<FontPackStatus>,
}

impl ExportPreview {
    #[must_use]
    pub fn has_errors(&self) -> bool {
        self.issues
            .iter()
            .any(|issue| issue.severity == ExportIssueSeverity::Error)
    }
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ExportProjectRequest {
    pub kind: ExportKind,
    pub destination: String,
    pub snapshot_token: String,
    pub overwrite: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum PdfStatus {
    NotRequested,
    Created,
    XeLatexMissing,
    Failed,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ExportResult {
    pub csv_path: Option<String>,
    pub latex_directory: Option<String>,
    pub zip_path: Option<String>,
    pub pdf_path: Option<String>,
    pub pdf_status: PdfStatus,
    pub row_count: usize,
    pub issues: Vec<ExportIssue>,
    pub diagnostic_path: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct TexEngineStatus {
    pub available: bool,
    pub path: Option<String>,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum FontPackState {
    Missing,
    Installed,
    Invalid,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct FontPackStatus {
    pub id: String,
    pub version: String,
    pub state: FontPackState,
    pub mandatory: bool,
    pub installed_bytes: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct WritingSystem {
    pub id: String,
    pub name: String,
    #[serde(rename = "type")]
    pub kind: String,
    pub script_code: Option<String>,
    pub language_tag: Option<String>,
    pub display_role: Option<String>,
    pub sort_order: i64,
    pub font_family: Option<String>,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct EntryForm {
    pub id: String,
    pub writing_system_id: String,
    pub text: String,
    pub variant_label: Option<String>,
    pub dialect: Option<String>,
    pub status: Option<String>,
    pub notes: Option<String>,
    pub sort_order: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ExampleForm {
    pub id: String,
    pub writing_system_id: String,
    pub text: String,
    pub sort_order: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct Example {
    pub id: String,
    pub translation: Option<String>,
    pub notes: Option<String>,
    pub sort_order: i64,
    pub forms: Vec<ExampleForm>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct Sense {
    pub id: String,
    pub gloss: Option<String>,
    pub definition: Option<String>,
    pub part_of_speech: Option<String>,
    pub semantic_domain: Option<String>,
    pub sort_order: i64,
    pub examples: Vec<Example>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct EntryRelation {
    pub id: String,
    pub target_entry_id: Option<String>,
    pub relation_type: String,
    pub fallback_text: Option<String>,
    pub notes: Option<String>,
    pub sort_order: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct LexicalEntry {
    pub id: String,
    pub notes: Option<String>,
    pub section_override: Option<String>,
    pub revision: i64,
    pub created_at: String,
    pub updated_at: String,
    pub forms: Vec<EntryForm>,
    pub senses: Vec<Sense>,
    pub relations: Vec<EntryRelation>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct EntrySummary {
    pub id: String,
    pub primary_form: String,
    pub secondary_form: Option<String>,
    pub parts_of_speech: Vec<String>,
    pub revision: i64,
    pub section_label: Option<String>,
    pub manual_order_pending: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub enum EntrySortMode {
    Auto,
    Manual,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct EntrySortSettingsV1 {
    pub version: u8,
    pub mode: EntrySortMode,
    pub writing_system_id: String,
    pub alphabet: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(tag = "kind", rename_all = "camelCase")]
pub enum ManualSortItem {
    Heading { id: String, label: String },
    Entry { entry_id: String },
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ManualSortLayoutV1 {
    pub version: u8,
    pub items: Vec<ManualSortItem>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ProjectSnapshot {
    pub root_path: String,
    pub project: Project,
    pub writing_systems: Vec<WritingSystem>,
    pub part_of_speech_options: Vec<String>,
    pub semantic_domain_options: Vec<String>,
    pub export_settings: ExportSettingsV1,
    pub entry_sort_settings: EntrySortSettingsV1,
    pub manual_sort_layout: ManualSortLayoutV1,
    pub entries: Vec<EntrySummary>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateProjectRequest {
    pub parent_dir: String,
    pub name: String,
    pub language_name: Option<String>,
    pub language_code: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateProjectSettingsRequest {
    pub name: String,
    pub language_name: Option<String>,
    pub language_code: Option<String>,
    pub analysis_language: Option<String>,
    pub description: Option<String>,
    pub writing_systems: Vec<WritingSystem>,
    pub part_of_speech_options: Vec<String>,
    pub semantic_domain_options: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SaveEntryRequest {
    pub entry: LexicalEntry,
    pub expected_revision: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DeleteEntryRequest {
    pub id: String,
    pub expected_revision: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct DeletedEntry {
    pub id: String,
    pub deleted_at: String,
}
